woodwork/statistics_utils/frequency_inference/_build_freq_dataframe.py

```diff
@@ -0,0 +1,13 @@
+from ._constants import CANDIDATE_COLUMN_NAME
+
+
+def _build_freq_dataframe(df):
+    def to_series(row):
+        for alias in row[CANDIDATE_COLUMN_NAME]:
```

Comment:
We only care about the most likely frequency right? I think we can trim down on the size of this dataframe (and maybe total run time) if we only determine which rows belong to a window where the most likely frequency was found.

Reply:
Also, I think this is only created to determine the "true" start and end dates according to the most likely frequency. I wonder if we can do that by just iterating over `candidate_df` and not having to create another dataframe.

Reply:
@freddyaboulton very good points! thanks

Reply:
I was able to remove this completely and generate bounds for a given alias during the windowing.